osu.Game/Overlays/Options/SliderOption.cs

```diff
@@ -1,53 +1,163 @@
-﻿using System;
+﻿//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using OpenTK;
+using OpenTK.Input;
 using OpenTK.Graphics;
+using osu.Framework.Allocation;
+using osu.Framework.Audio;
+using osu.Framework.Audio.Sample;
 using osu.Framework.Configuration;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
 using osu.Framework.Graphics.Primitives;
 using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Transformations;
 using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input;
+
+namespace osu.Game.Overlays.Options
+{
+    public class SliderOption<T> : FlowContainer where T : struct
+    {
+        private SliderBar<T> slider;
+        private SpriteText text;
+    
+        public string LabelText
+        {
+            get { return text.Text; }
+            set
+            {
+                text.Text = value;
+                text.Alpha = string.IsNullOrEmpty(value) ? 0 : 1;
+            }
+        }
+        
+        public BindableNumber<T> Bindable
+        {
+            get { return slider.Bindable; }
+            set { slider.Bindable = value; }
+        }
+
+        public SliderOption()
+        {
+            Direction = FlowDirection.VerticalOnly;
+            RelativeSizeAxes = Axes.X;
+            AutoSizeAxes = Axes.Y;
+            Children = new Drawable[]
+            {
+                text = new SpriteText { Alpha = 0 },
+                slider = new OsuSliderBar<T>
+                {
+                    Margin = new MarginPadding { Top = 5 },
+                    RelativeSizeAxes = Axes.X,
+                }
+            };
+        }
+
+        private class OsuSliderBar<U> : SliderBar<U> where U : struct
+        {
+            private AudioSample sample;
+            private double lastSample;
+            
+            private Container nub;
+            private Box leftBox, rightBox;
+            public OsuSliderBar()
+            {
+                Height = 22;
+                Children = new Drawable[]
+                {
+                    leftBox = new Box
+                    {
+                        Height = 2,
+                        RelativeSizeAxes = Axes.None,
+                        Anchor = Anchor.CentreLeft,
+                        Origin = Anchor.CentreLeft,
+                        Colour = new Color4(255, 102, 170, 255),

```

Comment:
I do not have all of the colors, but I can work with the colors we're already using in a seperate PR